### PR TITLE
LPS-150134 - Widget scopes not correctly applied when staging

### PR DESCRIPTION
**Pull request by DaniSzimko**
[LPS-150134](https://liferay.atlassian.net/browse/LPS-150134) - When checkStagingGroup is true and the portletScopeGroup is a staging group, set the scopeGroupId to the portletScopeGroupId + modify the relevant test class to run the Blog portlet tests

Hello team,
I would like to request a review of this PR, as it changes the getScopeGroupId method in the PortalImpl class
to align the behavior of the portlet's groupId with the approach outlined in the StagingGetScopeGroupIdTest class's _getExpectedGroup method.

from https://github.com/liferay-core-infra/liferay-portal/pull/5311

I reviewed the code and retested based on the reproduction steps mentioned in the ticket's description, and I could not reproduce the issue. The integration tests are also passing.

### CODE REVIEW

**Comment by dantewang** [comment]
@brunofalc Seems to be a TS PR.

**Comment by catarineaguiar** [comment]
ci:test:relevant

**Comment by catarineaguiar** [comment]
no regressions found. Bug is not reproducible

File: portal-impl/src/com/liferay/portal/util/PortalImpl.java [hunk 1]
```diff
@@ -4424,16 +4424,29 @@ else if (doAsGroup != null) {
...
+									portletScopeLiveGroup.getGroupId();
+							}
+							else {
+								scopeGroupId = portletScopeGroupId;
+							}
+						}
+						else {
+							scopeGroupId = portletScopeGroup.getGroupId();
+						}
 					}
```
Comment (yuhai) [inline portal-impl/src/com/liferay/portal/util/PortalImpl.java:4450]: Hi @DaniSzimko，for the fix, it seems a little complexed. We can simplify the logic. After `long portletScopeGroupId = _getScopeGroupId()`,  two else blocks actually return the same `portletScopeGroupId`. We may simplify it as the foolowing:
```
scopeGroupId = _getScopeGroupId(themeDisplay, layout, portletId);

					if (!checkStagingGroup && scopeGroupId != layout.getGroupId()) {
						Group portletScopeGroup =
								GroupLocalServiceUtil.fetchGroup(scopeGroupId);

						if (portletScopeGroup.isStagingGroup()) {
							Group portletScopeLiveGroup =
								portletScopeGroup.getLiveGroup();

							scopeGroupId = portletScopeLiveGroup.getGroupId();
						}
					}
					else if (checkStagingGroup &&
							 !liveGroup.isStagedRemotely()) {

						Group stagingGroup = liveGroup.getStagingGroup();

						scopeGroupId = stagingGroup.getGroupId();
					}
					else {
						scopeGroupId = liveGroup.getGroupId();
					}
```

But it seems also be not good.

**Comment by yuhai** [comment]
Hi @DaniSzimko,

I didn't finish the review completely yet. Just want to give some updates.

1. BaseLocalStagingTestCase class changes are not necessary. Because @DeleteAfterTestRun can delete data completely. From UI, deleting one site needn't to firstly disable staging.
2. Please see the above comment about the fix part.

I will continue to check the pr and need to further adjust the fix.

File: modules/apps/staging/staging-test/src/testIntegration/java/com/liferay/staging/test/StagingGetScopeGroupIdTest.java [hunk 2]
```diff
@@ -64,27 +62,31 @@ public void setUp() throws Exception {
...
 		_testGetScopeGroupId(true, true, BlogsPortletKeys.BLOGS);
 	}
 
 	@Override
 	protected String[] getNotStagedPortletIds() {
-		return new String[] {BlogsPortletKeys.BLOGS_ADMIN};
+		return new String[] {
+			BlogsPortletKeys.BLOGS_ADMIN, BlogsPortletKeys.BLOGS
+		};
 	}
```
Comment (yuhai) [inline modules/apps/staging/staging-test/src/testIntegration/java/com/liferay/staging/test/StagingGetScopeGroupIdTest.java:90]: @DaniSzimko, `BlogsPortletKeys.BLOGS_ADMIN` seems to be unnecessary, please remove it.

**Comment by yuhai** [comment]
Hi @DaniSzimko,

I just finish my review. For the fix part, because it will firstly `_getScopeGroupId(themeDisplay, layout, portletId)`, its return value must > 0 because layout and portletId are not NULL. So we need to adjust the previous logic. Please see the followings:
```
scopeGroupId = _getScopeGroupId(
						themeDisplay, layout, portletId);

					Group scopeGroup =
						GroupLocalServiceUtil.fetchGroup(scopeGroupId);

					if (checkStagingGroup) {
						if (!scopeGroup.isStagingGroup() && !scopeGroup.isStagedRemotely()) {
							Group stagingGroup = scopeGroup.getStagingGroup();

							scopeGroupId = stagingGroup.getGroupId();
						}
					}
					else {
						if (scopeGroup.isStagingGroup()) {
							Group portletScopeLiveGroup =
								scopeGroup.getLiveGroup();

							scopeGroupId = portletScopeLiveGroup.getGroupId();
						}
					}
```
Instead of using this [codes](https://github.com/yuhai/liferay-portal/blob/LPS-122481/portal-impl/src/com/liferay/portal/util/PortalImpl.java#L4427-L4447). Please check if it is good to you.

In addition, the test didn't cover the part https://github.com/liferay-core-infra/liferay-portal/blob/7dd4847401c500a877cd66e39cf265f0c5968266/modules/apps/staging/staging-test/src/testIntegration/java/com/liferay/staging/test/StagingGetScopeGroupIdTest.java#L97-L101. Please add the related tests.

Please see the above comments and resend it. By the way, for the `checkStagingGroup` expected behavior, the fix referred to the `StagingGetScopeGroupIdTest#_getExpectedGroup()`,  could you please confirm the behavior with Staging team if this is expected behavior for now?

Please let me know if you have further concerns.

Thanks,
Hai

**Pull request closed without merging**